fetch.bs

```diff
@@ -7234,69 +7250,75 @@ of running <a for=Body>consume body</a> with <a>this</a> and the following step
+<!-- This is wrong. It should really set an underlying concept to this. -->
 </div>
 
 <div algorithm>
 <p>The <dfn method for=Body><code>formData()</code></dfn> method steps are to return the result of
 running <a for=Body>consume body</a> with <a>this</a> and the following step given a
-<a for=/>byte sequence</a> <var>bytes</var>: switch on <a>this</a>'s <a for=Body>MIME type</a>'s
-<a for="MIME type">essence</a> and run the corresponding steps:
+<a for=/>byte sequence</a> <var>bytes</var>:
 
-<dl class=switch>
- <dt>"<code>multipart/form-data</code>",
- <dd>
-  <ol>
-   <li>
-    <p>Parse <var>bytes</var>, using the value of the `<code>boundary</code>` parameter from
-    <a>this</a>'s <a for=Body>MIME type</a>, per the rules set forth in
-    <cite>Returning Values from Forms: multipart/form-data</cite>. [[!RFC7578]]</p>
-
-    <p>Each part whose `<code>Content-Disposition</code>` header contains a `<code>filename</code>`
-    parameter must be parsed into an <a for="entry list">entry</a> whose value is a {{File}} object
-    whose contents are the contents of the part. The {{File/name}} attribute of the {{File}} object
-    must have the value of the `<code>filename</code>` parameter of the part. The {{Blob/type}}
-    attribute of the {{File}} object must have the value of the `<code>Content-Type</code>` header
-    of the part if the part has such header, and `<code>text/plain</code>` (the default defined by
-    [[!RFC7578]] section 4.4) otherwise.</p>
-
-    <p>Each part whose `<code>Content-Disposition</code>` header does not contain a
-    `<code>filename</code>` parameter must be parsed into an <a for="entry list">entry</a> whose
-    value is the <a lt="UTF-8 decode without BOM">UTF-8 decoded without BOM</a> content of the part.
-    <span class=note>This is done regardless of the presence or the value of a
-    `<code>Content-Type</code>` header and regardless of the presence or the value of a
-    `<code>charset</code>` parameter.</span></p>
-
-    <p class=note>A part whose `<code>Content-Disposition</code>` header contains a
-    `<code>name</code>` parameter whose value is `<code>_charset_</code>` is parsed like any other
-    part. It does not change the encoding.</p>
-   </li>
-
-   <li><p>If that fails for some reason, then <a>throw</a> a {{TypeError}}.
-
-   <li><p>Return a new {{FormData}} object, appending each <a for="entry list">entry</a>, resulting
-   from the parsing operation, to its <a for=FormData>entry list</a>.
-  </ol>
+<ol>
+ <li><p>Let <var>mimeType</var> be the result of <a for=Body>get the MIME type</a>
+ with <a>this</a>.
 
-  <p class=XXX>The above is a rough approximation of what is needed for
-  `<code>multipart/form-data</code>`, a more detailed parsing specification is to be
-  written. Volunteers welcome.
+ <li><p>If <var>mimeType</var> is non-null, then switch on <var>mimeType</var>'s
+ <a for="MIME type">essence</a> and run the corresponding steps:
```

Comment:
Since this `<li>` contains both the `<p>` and the `<dl>`, the `<p>` needs to be on a newline and indented once. The `<dl>` needs to be indented one more as well therefore.

Wording-wise this looks great though!